demos/dev_mode_key_provisioning/include/aws_dev_mode_key_provisioning.h

```diff
@@ -23,87 +23,192 @@
                              uint8_t * pucLabel,
-                             CK_OBJECT_HANDLE_PTR xObjectHandle );
+                             CK_OBJECT_HANDLE_PTR pxObjectHandle );
+
+/** \brief Generates an RSA key pair.
+ *
+ * \param[in] xSession              A valid PKCS #11 session handle.
+ * \param[in] pucPrivateKeyLabel    PKCS #11 label attribute value for private key to be created.
+ *                                  This should be a string value. See iot_pkcs11_config.h.
+ * \param[in] pucPublicKeyLabel     PKCS #11 label attribute value for public key to be created.
+ *                                  This should be a string value. See iot_pkcs11_config.h.
+ * \param[in] pxPrivateKeyHandle    Updated to provide the PKCS #11 handle for the private key
+ *                                  that was generated.
+ * \param[out] pxPublicKeyHandle    Updated to provide the PKCS #11 handle for the public key
+ *                                  that was generated.
+ *
+ * \return CKR_OK if RSA key pair generation succeeded.
+ * Otherwise, a positive PKCS #11 error code.
+ */
+CK_RV xProvisionGenerateKeyPairRSA( CK_SESSION_HANDLE xSession,
+                                    uint8_t * pucPrivateKeyLabel,
+                                    uint8_t * pucPublicKeyLabel,
+                                    CK_OBJECT_HANDLE_PTR pxPrivateKeyHandle,
+                                    CK_OBJECT_HANDLE_PTR pxPublicKeyHandle );
+
+/** \brief Generates an elliptic curve key pair.
+ *
+ * \param[in] xSession              A valid PKCS #11 session handle.
+ * \param[in] pucPrivateKeyLabel    PKCS #11 label attribute value for private key to be created.
+ *                                  This should be a string value. See iot_pkcs11_config.h.
+ * \param[in] pucPublicKeyLabel     PKCS #11 label attribute value for public key to be created.
+ *                                  This should be a string value. See iot_pkcs11_config.h.
+ * \param[in] pxPrivateKeyHandle    Updated to provide the PKCS #11 handle for the private key
+ *                                  that was generated.
+ * \param[out] pxPublicKeyHandle    Updated to provide the PKCS #11 handle for the public key
+ *                                  that was generated.
+ *
+ * \return CKR_OK if EC key pair generation succeeded.
+ * Otherwise, a positive PKCS #11 error code.
+ */
+CK_RV xProvisionGenerateKeyPairEC( CK_SESSION_HANDLE xSession,
+                                   uint8_t * pucPrivateKeyLabel,
+                                   uint8_t * pucPublicKeyLabel,
+                                   CK_OBJECT_HANDLE_PTR pxPrivateKeyHandle,
+                                   CK_OBJECT_HANDLE_PTR pxPublicKeyHandle );
+
+/** \brief Destroys AFR credentials stored in device PKCS #11 module.
+*
+* \note Not all ports support the deletion of all objects.  Successful
+* function return only indicates that all objest for which destroy is 
```

Comment:
"Objects". 

Is this behavior per spec? Seems potentially annoying for the app writer, if so.